src/python/arcor2/cached.py

```diff
@@ -431,6 +437,58 @@ def constant(self, constant_id: str) -> cmn.ProjectConstant:
         except KeyError:
             raise CachedProjectException("Constant not found.")
 
+    def get_by_id(
+        self, obj_id: str
+    ) -> Union[cmn.SceneObject, cmn.BareActionPoint, cmn.NamedOrientation, cmn.ProjectRobotJoints, cmn.Action]:
+
+        if obj_id in self._action_points:  # AP
+            return self._action_points[obj_id]
+        elif obj_id in self._joints.data:  # Joints
+            return self._joints.data[obj_id]
+        elif obj_id in self._orientations.data:  # Orientation
+            return self._orientations.data[obj_id]
+        elif obj_id in self._actions.data:  # Action
+            return self._actions.data[obj_id]
+
+        from arcor2_arserver.globals import LOCK
```

Comment:
Tady by se určitě nemělo importovat nic z arserveru. To je vlastně kruhová závislost - sice obejitá tím že je import v metodě, ale nemělo by to být.

Reply:
Metoda by se asi víc hodila do třídy Lock, aby mohla vracet i objekty scény, o kterých projekt nemůže vědět.